yt/data_objects/data_containers.py

```diff
@@ -1673,8 +1664,8 @@ def to_frb(self, width, resolution, center=None, height=None,
                     "Currently we only support images centered at R=0. " +
                     "We plan to generalize this in the near future")
             from yt.visualization.fixed_resolution import CylindricalFixedResolutionBuffer
-            if iterable(width):
-                radius = max(width)
+            if isinstance(width, tuple):
+               radius = width[0]
```

Comment:
I don't think this change is correct - the original code is correct. We're checking for `iterable(width)` because width might be a list or an ndarray or some other kind of iterable. It's easier to just use the `iterable` function to test for this rather than using `isinstance` and insisting on only accepting a certain number of types.

Also it's a behavior change to pick `width[0]` instead of `max(width)`. Can you explain why you did that? Choosing `max(width)` is arbitrary, but since we already made the decision it's best to not change our minds without a good reason.

Reply:
I changed it as I was getting error for the test case:
```
def test_to_frb():
    ds = fake_amr_ds(fields=["density", "cell_mass"], geometry="cylindrical",
                     particles=16**3)
    proj = ds.proj("density", weight_field="cell_mass", axis=1,
                   data_source=ds.all_data())
    proj.to_frb((1.0, 'unitary'), 64)
```
Error:
```
line 1668, in to_frb
    radius = max(width)
TypeError: '>' not supported between instances of 'str' and 'float'
```

Reply:
Width is allowed to be a tuple of tuples, like this:

```
((1.0, 'unitary'), (0.7, 'unitary'))
```

this allows constructing images where the width doesn't equal the height.

For the cylindrical FRB the width really represents a radius, and what happens right now is it assumes if you're passing a tuple you're passing two floats (which probably needs to be relaxed) and it assumes the radius you *really* mean is the maximum of the two floats you passed in.

That's not great, so maybe it's ok to change this. Rather than what you've done though I'd suggest if we're passed a tuple, to check if it's a valid `(length, unit)` tuple, and if not raise an error. That is, for a cylindrical dataset, it only makes sense to pass in one width to inrepret as the radius, so if someone passes in more than one, we should raise an error.

Basically the code here:

https://github.com/yt-project/yt/blob/cc299b07a5923b28c9105bd898e231412f25d3cc/yt/data_objects/data_containers.py#L1691-L1696

But in the if statement raise an error instead of interpreting the tuple as a width and height.